Upstream: Yes
From bb7f58f5f88381857cd9b5dbe608e75d459873f1 Mon Sep 17 00:00:00 2001
From: Matthias Klumpp <matthias@tenstral.net>
Date: Tue, 24 Mar 2020 18:25:02 +0100
Subject: [PATCH] Mark featured AppStream snippets as append-merge components -
 again

This once again fixes the issue that these components are otherwise
considered invalid new components, instead of modifications to existing
components (and this will produce an annoying warning message, asking
the distributor to fix their data).
---
 data/assets/org.gnome.Software.Featured.xml | 58 ++++++++++-----------
 1 file changed, 29 insertions(+), 29 deletions(-)

diff --git a/data/assets/org.gnome.Software.Featured.xml b/data/assets/org.gnome.Software.Featured.xml
index 92576d2c..d1d920be 100644
--- a/data/assets/org.gnome.Software.Featured.xml
+++ b/data/assets/org.gnome.Software.Featured.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <components>
-  <component>
+  <component merge="append">
     <id>inkscape.desktop</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border-color: #aaa;
@@ -13,7 +13,7 @@ background:
 </value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>mypaint.desktop</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border-color: #888;
@@ -30,7 +30,7 @@ background:
 </value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>org.gnome.Maps.desktop</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border-color: #deddda;
@@ -45,7 +45,7 @@ background: #f6f5f4,
 </value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>org.gnome.Polari.desktop</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border-color: #4e9a06;
@@ -60,7 +60,7 @@ background:
 </value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>transmission-gtk.desktop</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border-color: #d65100;
@@ -74,7 +74,7 @@ background:
 </value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>org.gnome.Podcasts</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid #d7430a;
@@ -89,7 +89,7 @@ text-shadow: 0 1px 12px rgba(0,0,0,0.05),
       -1px 0 rgba(0,0,0,0.1);</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>org.gnome.Boxes.desktop</id>
     <categories>
       <category>Featured</category>
@@ -106,7 +106,7 @@ text-shadow: 0 1px 12px rgba(0,0,0,0.05),
       -1px 0 rgba(0,0,0,0.1);</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>com.github.johnfactotum.Foliate</id>
     <categories>
       <category>Featured</category>
@@ -123,7 +123,7 @@ text-shadow: none;
 </value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>org.gnome.SoundRecorder.desktop</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid #bbb;
@@ -133,7 +133,7 @@ color: #fff;
 text-shadow: none;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>org.kde.kdenlive</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid #df321e;
@@ -144,7 +144,7 @@ color: #df321e;
 text-shadow: none;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>org.gimp.GIMP</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border-color: #4a8c30;
@@ -157,7 +157,7 @@ background:
 </value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>org.videolan.VLC</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid #df321e;
@@ -168,7 +168,7 @@ color: #222;
 text-shadow: 0 0 2px #fff;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>com.spotify.Client</id>
     <categories>
       <category>Featured</category>
@@ -181,7 +181,7 @@ background: url('http://people.gnome.org/~jimmac/gnome-software/spotify-logo.svg
  10% / 200px auto no-repeat, -gtk-gradient (linear, left top, left bottom, from(#b47fba), to(#000));</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>com.slack.Slack</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid #eee;
@@ -194,7 +194,7 @@ color: #fff;
 text-shadow: none;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>org.gnome.FeedReader</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border-color: #d65100;
@@ -206,7 +206,7 @@ background: #f5c211
 </value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>com.bitwarden.desktop</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid #1c6d8c;
@@ -217,7 +217,7 @@ color: #fff;
 text-shadow: 0 0 2px #26a269;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>com.dropbox.Client</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid #a8c3d6;
@@ -227,7 +227,7 @@ background: #b8d3e6 url('file://@datadir@/gnome-software/featured-dropbox.png')
 color: #47525d;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>com.github.calo001.fondo</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid transparent;
@@ -238,7 +238,7 @@ color: #fff;
 text-shadow: 0 1px 2px #000;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>org.darktable.Darktable</id>
     <categories>
       <category>Featured</category>
@@ -251,7 +251,7 @@ color: #fff;
 text-shadow: 0 1px 2px #000;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>com.calibre_ebook.calibre</id>
     <categories>
       <category>Featured</category>
@@ -264,7 +264,7 @@ color: #fff;
 text-shadow: 0 1px 2px #000;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>fr.free.Homebank</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid transparent;
@@ -274,7 +274,7 @@ color: #fff;
 text-shadow: 0 1px 2px #000;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>de.wolfvollprecht.UberWriter</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid #0554c4;
@@ -285,7 +285,7 @@ color: #fff;
 text-shadow: 0 0 3px #3584e4;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>net.sourceforge.Klavaro</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid transparent;
@@ -295,7 +295,7 @@ color: #000;
 text-shadow: 0 0 4px #fff;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>com.github.junrrein.PDFSlicer</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid #1c6d8c;
@@ -306,7 +306,7 @@ color: #fff;
 text-shadow: none;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>org.gnome.Todo.desktop</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid transparent;
@@ -316,7 +316,7 @@ color: #fff;
 text-shadow: 0 1px 2px #000;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>com.github.fabiocolacio.marker</id>
     <categories>
       <category>Featured</category>
@@ -329,7 +329,7 @@ color: #fff;
 text-shadow: 0 1px 2px #000;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>com.valvesoftware.Steam</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border-color: #1b2838;
@@ -339,7 +339,7 @@ background: #1b2838 url('http://people.gnome.org/~jimmac/gnome-software/steam-lo
  10% / 200px auto no-repeat;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>org.gnome.Tetravex</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid transparent;
@@ -349,7 +349,7 @@ color: #000;
 text-shadow: 0 0 2px #fff;</value>
     </custom>
   </component>
-  <component>
+  <component merge="append">
     <id>org.stellarium.Stellarium</id>
     <custom>
       <value key="GnomeSoftware::FeatureTile-css">border: 1px solid transparent;
-- 
2.24.1

